File: ios/ReactNativeBrownfieldModule.m
#import "ReactNativeBrownfieldModule.h"

#if __has_include("ReactNativeBrownfield/ReactNativeBrownfield-Swift.h")
#import "ReactNativeBrownfield/ReactNativeBrownfield-Swift.h"
#else
#import "ReactNativeBrownfield-Swift.h"
#endif

@implementation ReactNativeBrownfieldModule

RCT_EXPORT_MODULE(ReactNativeBrownfield);

RCT_EXPORT_METHOD(setPopGestureRecognizerEnabled:(BOOL)enabled) {
  [ReactNativeBrownfieldModuleImpl setPopGestureRecognizerEnabled:enabled];
}

RCT_EXPORT_METHOD(popToNative:(BOOL)animated) {
  [ReactNativeBrownfieldModuleImpl popToNativeWithAnimated:animated];
}

@end

File: ios/ReactNativeBrownfieldModule.swift
internal import React

@objcMembers
public class ReactNativeBrownfieldModuleImpl: NSObject {
  static public func setPopGestureRecognizerEnabled(_ enabled: Bool) {
    let userInfo = ["enabled": enabled]
    DispatchQueue.main.async {
      NotificationCenter.default.post(name: Notification.Name.togglePopGestureRecognizer, object: nil, userInfo: userInfo)
    }
  }

  static public func popToNative(animated: Bool, result: [String: Any]?) {
    var userInfo: [String : Any] = ["animated": animated]
    if result != nil {
      userInfo["result"] = result
    }
    DispatchQueue.main.async {
      NotificationCenter.default.post(name: Notification.Name.popToNative, object: nil, userInfo: userInfo)
    }
  }
}
